### ISSUE TITLE
Parsing error with accented characters

### ISSUE THREAD

**Issue by alexandrethm** [sqlfluff/sqlfluff#2655]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have SQL file with some accented characters in column names : `à`, `é`, `è`, etc (which is a terrible idea, I know, unfortunately I have to work with this data) and I cannot get it parsed with `sqlfluff`.

Here is an example `test.sql` file:
```sql
select
    col_one,
    col_two,

    col_with_à_special_char
from table;
```

### Expected Behaviour

I would expect sqlfluff to correctly parse and format the file, e.g.
```sql
select
    col_one,
    col_two,
    col_with_à_special_char
from table;
```
(dbt actually handles these accented characters correctly)

### Observed Behaviour

Running `sqlfluff fix test.sql` actually gives the following:
```
==== finding fixable violations ====
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [1 unfixable linting violations found]
```

I tried adding explicitely the option `--encoding utf-8`, which helped sqlfluff recognize characters correctly (and display `à` instead of `Ã\xa0` in the verbose output), but linting still fails.

Here is the full verbose output (`sqlfluff lint -v -v -v --encoding utf-8 test.sql`):
<details>

```
==== sqlfluff ====
sqlfluff:               0.10.0 python:                 3.8.11
implementation:        cpython dialect:                  ansi
verbosity:                   3 templater:               jinja

== Raw Config:
core:
    dialect:            ansi
    disable_noqa:       False
    encoding:           utf-8
    fix_even_unparsable:False
    ignore:             []
    ignore_templated_areas:True
    nocolor:            False
    output_line_length: 80
    recurse:            True
    runaway_limit:      10
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl
    templater:          jinja
    verbose:            3
indentation:
    indented_ctes:      False
    indented_joins:     False
    indented_using_on:  True
    template_blocks_indent:True
rules:
    allow_scalar:       True
    comma_style:        trailing
    indent_unit:        space
    max_line_length:    80
    single_table_references:consistent
    tab_space_size:     4
    unquoted_identifiers_policy:all
    L007:
        operator_new_lines: after
    L010:
        capitalisation_policy:consistent
    L011:
        aliasing:           explicit
    L012:
        aliasing:           explicit
    L014:
        extended_capitalisation_policy:consistent
    L016:
        ignore_comment_clauses:False
        ignore_comment_lines:False
    L026:
        force_enable:       False
    L028:
        force_enable:       False
    L029:
        unquoted_identifiers_policy:aliases
    L030:
        capitalisation_policy:consistent
    L038:
        select_clause_trailing_comma:forbid
    L040:
        capitalisation_policy:consistent
    L042:
        forbid_subquery_in: join
    L047:
        prefer_count_0:     False
        prefer_count_1:     False
    L052:
        multiline_newline:  False
        require_final_semicolon:False
    L054:
        group_by_and_order_by_style:consistent
    L057:
        additional_allowed_characters:""
        allow_space_in_identifier:False
        quoted_identifiers_policy:all
        unquoted_identifiers_policy:all
    L059:
        prefer_quoted_identifiers:False
    L062:
templater:
    unwrap_wrapped_queries:True
    jinja:
        apply_dbt_builtins: True
        macros:
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_is_incremental: {% macro is_incremental() %}True{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable, default='') %}item{% endmacro %}

==== readout ====

=== [ path: test.sql ] ===

INFO       TEMPLATING RAW [jinja] (test.sql)
INFO       Slicing File Template
INFO       [L001] No config_keywords defined for L001
INFO       [L005] No config_keywords defined for L005
INFO       [L006] No config_keywords defined for L006
INFO       [L008] No config_keywords defined for L008
INFO       [L009] No config_keywords defined for L009
INFO       [L015] No config_keywords defined for L015
INFO       [L017] No config_keywords defined for L017
INFO       [L020] No config_keywords defined for L020
INFO       [L021] No config_keywords defined for L021
INFO       [L023] No config_keywords defined for L023
INFO       [L024] No config_keywords defined for L024
INFO       [L025] No config_keywords defined for L025
INFO       [L027] No config_keywords defined for L027
INFO       [L031] No config_keywords defined for L031
INFO       [L032] No config_keywords defined for L032
INFO       [L033] No config_keywords defined for L033
INFO       [L034] No config_keywords defined for L034
INFO       [L035] No config_keywords defined for L035
INFO       [L036] No config_keywords defined for L036
INFO       [L037] No config_keywords defined for L037
INFO       [L039] No config_keywords defined for L039
INFO       [L041] No config_keywords defined for L041
INFO       [L043] No config_keywords defined for L043
INFO       [L044] No config_keywords defined for L044
INFO       [L045] No config_keywords defined for L045
INFO       [L046] No config_keywords defined for L046
INFO       [L048] No config_keywords defined for L048
INFO       [L049] No config_keywords defined for L049
INFO       [L050] No config_keywords defined for L050
INFO       [L051] No config_keywords defined for L051
INFO       [L053] No config_keywords defined for L053
INFO       [L055] No config_keywords defined for L055
INFO       [L056] No config_keywords defined for L056
INFO       [L058] No config_keywords defined for L058
INFO       [L060] No config_keywords defined for L060
INFO       [L061] No config_keywords defined for L061
INFO       LEXING RAW (test.sql)
INFO       Elements to Segments.
INFO       Source-only slices: []
INFO       Lexed tokens: ['select', '\n', '    ', 'col_one', ',', '\n', '    ', 'col_two', ',', '\n', '\n', '    ', 'col_with_', 'à_special_char', '\n', 'from', ' ', 'table', ';', '\n']
INFO       PARSING (test.sql)
INFO
###
#
# Parsed Tree:
#
###
INFO
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    column_reference:
[L:  2, P:  5]      |                        identifier:                           'col_one'
[L:  2, P: 12]      |                comma:                                        ','
[L:  2, P: 13]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                select_clause_element:
[L:  3, P:  5]      |                    column_reference:
[L:  3, P:  5]      |                        identifier:                           'col_two'
[L:  3, P: 12]      |                comma:                                        ','
[L:  3, P: 13]      |                newline:                                      '\n'
[L:  4, P:  1]      |                newline:                                      '\n'
[L:  5, P:  1]      |                whitespace:                                   '    '
[L:  5, P:  5]      |                select_clause_element:
[L:  5, P:  5]      |                    column_reference:
[L:  5, P:  5]      |                        identifier:                           'col_with_'
[L:  5, P: 14]      |                    unparsable:                               !! Expected: 'Nothing...'
[L:  5, P: 14]      |                        unlexable:                            'à_special_char'
[L:  5, P: 28]      |            newline:                                          '\n'
[L:  6, P:  1]      |            [META] dedent:
[L:  6, P:  1]      |            from_clause:
[L:  6, P:  1]      |                keyword:                                      'from'
[L:  6, P:  5]      |                whitespace:                                   ' '
[L:  6, P:  6]      |                from_expression:
[L:  6, P:  6]      |                    [META] indent:
[L:  6, P:  6]      |                    from_expression_element:
[L:  6, P:  6]      |                        table_expression:
[L:  6, P:  6]      |                            table_reference:
[L:  6, P:  6]      |                                identifier:                   'table'
[L:  6, P: 11]      |                    [META] dedent:
[L:  6, P: 11]      |    statement_terminator:                                     ';'
[L:  6, P: 12]      |    newline:                                                  '\n'

INFO       Found unparsable segment...
INFO       [L:  5, P: 14]      |unparsable:                                                   !! Expected: 'Nothing...'
[L:  5, P: 14]      |    unlexable:                                                'à_special_char'

INFO       LINTING (test.sql)
== [test.sql] LINTING
== [test.sql] FAIL
L:   5 | P:   5 | L013 | Column expression without alias. Use explicit `AS`
                       | clause.
L:   5 | P:  14 |  LXR | Unable to lex characters: 'à_special_...'
L:   5 | P:  14 |  PRS | Line 5, Position 14: Found unparsable section:
                       | 'à_special_char'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
==== summary ====
files:             1 violations:        3
clean files:       0 unclean files:     1
avg per file:   3.00 unclean rate:   100%
status:         FAIL
All Finished 📜 🎉!
```

</details>

### How to reproduce

Use `test.sql` example above

### Dialect

default `ansi`

### Version

`sqlfluff, version 0.10.0`

### Configuration

nothing changed

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
What DBMS are you using? As most of them expect accepted characters to be quoted.

This works, when parsed as ANSI:

```sql
select
    col_one,
    col_two,
    'col_with_à_special_char'
from table;
```

Whether it should work without quotes depends on the dialect. ANSI is not a real dialect so need to know what dialect you're actually using.

**Comment by tunetheweb** [comment]
> Oh I see, I am using `postgres`, which seemed to work without quotes. Adding quotes does indeed fix the issue, so thank you!

Your original issue said you were using ansi dialect.

It looks like using the unquoted form [is valid postgres](https://www.postgresql.org/docs/14/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS)

> SQL identifiers and key words must begin with a letter (a-z, but also letters with diacritical marks and non-Latin letters) or an underscore (_). Subsequent characters in an identifier or key word can be letters, underscores, digits (0-9), or dollar signs ($). Note that dollar signs are not allowed in identifiers according to the letter of the SQL standard, so their use might render applications less portable. The SQL standard will not define a key word that contains digits or starts or ends with an underscore, so identifiers of this form are safe against possible conflict with future extensions of the standard.

So should we reopen this if you would like that to b e supported?

**Comment by alexandrethm** [comment]
> Your original issue said you were using ansi dialect. 
Yes indeed, sorry ! I did not think the dialect would actually matter, so I tried with ansi as well in this simple example (not knowing about the quotes specification).

> So should we reopen this if you would like that to be supported?
I personally do not mind adding double quotes around the column names with accented characters 👌
